instrumentation/jmx-metrics/library/build.gradle.kts

```diff
@@ -6,4 +8,28 @@ dependencies {
   implementation("org.snakeyaml:snakeyaml-engine")
 
   testImplementation(project(":testing-common"))
+  testImplementation("org.testcontainers:testcontainers")
+
+  testImplementation("org.testcontainers:junit-jupiter")
+  testImplementation("com.linecorp.armeria:armeria-junit5:1.31.3")
+  testImplementation("com.linecorp.armeria:armeria-junit5:1.31.3")
+  testImplementation("com.linecorp.armeria:armeria-grpc:1.31.3")
+  testImplementation("io.opentelemetry.proto:opentelemetry-proto:1.5.0-alpha")
+}
+
+tasks {
+  test {
+    // get packaged agent jar for testing
+    val shadowTask = project(":javaagent").tasks.named<ShadowJar>("shadowJar").get()
+
+    dependsOn(shadowTask)
+
+    inputs.files(layout.files(shadowTask))
+      .withPropertyName("javaagent")
+      .withNormalizer(ClasspathNormalizer::class)
+
+    doFirst {
+      jvmArgs("-Dio.opentelemetry.javaagent.path=${shadowTask.archiveFile.get()}")
+    }
```

Comment:
[for reviewer] using similar strategy we have in smoke tests to get a copy of the packaged agent jar.